### DISCUSSION TITLE
Can multiple useContractRead calls be combined into a single multicall?

### DISCUSSION THREAD

**Discussion by joshlangs** [wevm/wagmi#1883] [Q&A]
Using a blank create-wagmi (Vite React + Default) app, 

I add some code like this:
```
useContractRead({ abi: erc20ABI, address: "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48", functionName: "balanceOf", args: ["0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48"] }),
useContractRead({ abi: erc20ABI, address: "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48", functionName: "balanceOf", args: ["0xdAC17F958D2ee523a2206206994597C13D831ec7"] }),
...
...a bunch more, all with different args
...
```

It causes many `eth_call` api calls, and none to the `multicall` address

Could I ask what's the correct approach to getting these all batched into a single multicall?  Ideally, batch all `useContractRead` calls that happen during a render across all my components) into a single multicall?

**Comment by jxom** [comment]
Currently, you'd need to hoist them into [`useContractReads`](https://wagmi.sh/docs/hooks/useContractReads) to benefit from `multicall`.

Maybe in the future (shortly after wagmi@1), we will look into batching up `useContractRead`'s into a multicall scheduler.

**Comment by joshlangs** [comment]
Ahh, fantastic that it's coming.

It's really a necessary use-case.  I was in the middle of porting an app from useDapp -> wagmi, when I was notified that I'd hit 75% of my infura usage.  ...All by myself on my dev machine.  That's when I discovered that 50-100 calls were going out every few seconds.  So... the porting is paused for now

But I love what you guys are doing, and will port ASAP once batching is implemented
